### ISSUE TITLE
Slideshow/Image Preview not working in public share link.

### ISSUE THREAD

**Issue by imjalpreet** [owncloud/gallery#601]
### Steps to reproduce

As a logged in user,
1. Generate a public share link.
2. Go to that link.
3. Click on any image.
### Expected behaviour

BigShot container should show up, i.e., Image should show up in the BigShot container.
### Actual behaviour

Nothing happens and there is this error in the console: 
`Uncaught TypeError: Cannot read property 'hideDropDown' of undefined`
### Server configuration

**Operating system**: Ubuntu 14.04

**Web server:**Apache/2.4.7

**Database:** mysql Ver 14.14 Distrib 5.6.28, for debian-linux-gnu (x86_64)

**PHP version:** PHP 5.5.9-1ubuntu4.14

**ownCloud version:** 9.1.0 pre alpha

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - dav: 0.2.1
  - federatedfilesharing: 0.2.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_sharing: 0.10.0
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - gallery: 15.0.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.0
Disabled:
  - encryption
  - files_external
  - testing
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oc4z46udfiog",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost:8080"
        ],
        "datadirectory": "\/var\/www\/GSoC\/data",
        "overwrite.cli.url": "http:\/\/localhost:8080",
        "dbtype": "mysql",
        "version": "9.1.0.0",
        "dbname": "ownCloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "jalpreet.cse",
        "mail_smtpmode": "smtp",
        "mail_domain": "gmail.com",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "smtp.gmail.com",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0
    }
}
```

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using custom gallery.cnf config files:** No
#### ownCloud log (data/owncloud.log)

The log is over [here](https://gist.github.com/imjalpreet/2188043ccc6bcfcc487f)
### Client configuration

**Browser:** Chrome 49.0.2623.75 (64-bit)

**Operating system:** Ubuntu 14.04
### JavaScript Console Error

`Uncaught TypeError: Cannot read property 'hideDropDown' of undefined`

**Comment by oparoz** [comment]
Thanks.

@viraj96 - You need to take care of this issue as well. There is no sharing on the public side, so you need to make sure `Gallery.Share` is not `undefined` before calling `hideDropDown`.

**Comment by imjalpreet** [comment]
The error is over [here](https://github.com/owncloud/gallery/blob/master/js/slideshow.js#L95)

I think we just need to add a check: `if (Gallery.Share)` and if it is not undefined, then only do `Gallery.Share.hideDropDown()`.

**Comment by imjalpreet** [comment]
Sorry, I didn't read your previous comment. :+1:

**Comment by oparoz** [comment]
Fixed via #613
